tests/pytest_tests/system_tests/test_launch/test_launch_manage.py

```diff
@@ -0,0 +1,185 @@
+"""System tests for the `wandb.sdk.launch.inputs.manage` module.
+
+These tests execute the real user flow exposed by the
+`wandb.sdk.launch.inputs.manage` module. These tests
+
+"""
+
+import time
+
+import pytest
+import wandb
+import wandb.env
+import yaml
+from wandb.sdk import launch
+
+
+@pytest.fixture
+def test_config():
+    return {
+        "key1": 1,
+        "key2": {
+            "key3": "hello",
+            "key4": {
+                "key5": "world",
+                "key6": {
+                    "key7": 1,
+                    "key8": 2,
+                },
+            },
+        },
+    }
+
+
+@pytest.mark.skipif(not wandb.env.is_require_core(), reason="core")
+def test_manage_config_file(
+    test_settings,
+    wandb_init,
+    tmp_path,
+    test_config,
+    monkeypatch,
+):
+    """Test that calling `manage_config_file` has the expected side effects.
+
+    If `manage_config_file` is called and a job is created then the run should
+    have the config file saved in its files and the job should have the config
+    file schema.
+    """
+    monkeypatch.chdir(tmp_path)
+    config_str = yaml.dump(test_config)
+
+    (tmp_path / "config.yaml").write_text(config_str)
+    settings = test_settings()
+    settings.update(
+        {
+            "program_relpath": "./blah/test_program.py",
+            "git_remote_url": "https://github.com/test/repo",
+            "git_commit": "asdasdasdasd",
+        }
+    )
+    with wandb_init(settings=settings) as run:
+        launch.manage_config_file(
+            "config.yaml",
+            include=["key2"],
+            exclude=["key2.key4.key6.key8", "key2.key3"],
+        )
+        run.log({"test": 1})
+
+    time.sleep(5)  # Wait for run files to be retrievable from the server.
```

Comment:
Sleeps aren’t good, can we use the relay server to catch these requests and ensure they are correct?